### DISCUSSION TITLE
determine PGM status

### DISCUSSION THREAD

**Discussion by icygame** [ufodone/envisalink_new#130] [Q&A] [answered]
The new plugin (working very well) allows HA to activate a PGM output as per one of the other discussions above...I have essentially the inverse objective, to detect if a PGM has been activated (from the keypad). 

Wondering if this would be possible...my hunch is that at the very least it should be de able to be determined by parsing what is displayed in the keypad screen...but I can't see a service to do that??

(Context is the DSC color display offers 4 PGM 'buttons'... detecting activation would allow the keypad to be used for to trigger other HA automations eg locking door, closing gate. Obviously another approach would be to add a Shelly or some other sensor and a relay to the PGMs but that is more points of failure....

**Comment by ufodone** [comment, marked as answer]
I think you might be able to use a state trigger in your automation and watch for the `pgm_<n>_last_triggered` attribute to change on the keypad sensor.  That attribute should be updated any time a PGM is triggered.

A trigger watching for PGM 2 might look something like:

```
platform: state
entity_id:
  - sensor.home_alarm_keypad
attribute: pgm_2_last_triggered
```

I haven't tried this myself though as I don't have any PGMs setup on my own panel.

**Comment by icygame** [comment]
Thanks for the suggestion. I can see an attribute 'PGMrelay' but this always always reads 'true' so i assume that is just reporting that there is a PGM configured. I'd missed that the attribute 'alpha' which is the on-screen text does update when the PGM is triggered so I can use that!
cheers.

[but should i be seeing the attribute pgm_<n>_last_triggered...would be more elegant....but i can't see it as an attribute of the keypad sensor]

**Comment by ufodone** [comment]
That sounds like you are running an older version.  The [0.9.1](https://github.com/ufodone/envisalink_new/releases/tag/v0.9.1) release contains the changes I referred to above.  The `PGMrelay` attribute was removed because, as you determined, it always reported true.

It's possible the `pgm_<n>_last_triggered` might not show up initially until a PGM has actually been triggered but that shouldn't prevent you from configuring a trigger for it (it just wouldn't show up in the drop-down).

**Reply by icygame** [reply]
OKAY....I appear to be on 0.1 so that would be right!
However...is there anything odd about the versioning? I've gone through the motions of downloading to 0.9.1 (and downgrading from that to 0.9.0 and accepting the update recommendation), deleting the integration, multiple restarts etc...but the integration still reports version 0.1. I initially had the old YAML config remaining but have now deleted and reconfigured from scratch...still 0.1.

**Comment by ufodone** [comment]
> However...is there anything odd about the versioning? 

Yes, it's broken.  :smile:  I only realized recently that while I'd been versioning the releases here, I wasn't updating the version that shows up in HA.  So they will all show as 0.1 at the moment.  I'll fix that for future releases.

**Reply by icygame** [reply]
ah yeah i just saw the that in the manifest file. But the code should be correct? In which case i will need to delve a bit further.

**Comment by ufodone** [comment]
Well, I also somehow managed to not included those PGM changes in the last release.  😬 I've published a new 0.9.2 release which should both fix the version and include the changes that you need.  Hopefully I got it right this time...

**Comment by icygame** [comment]
Perfect.Sent from my iPhoneOn 14 Aug 2024, at 14:03, ufodone ***@***.***> wrote:﻿
Well, I also somehow managed to not included those PGM changes in the last release.  😬 I've published a new 0.9.2 release which should both fix the version and include the changes that you need.  Hopefully I got it right this time...

**Reply by icygame** [reply]
We got there! Working properly, thanks. And cleaned up a few things on the way!!
